### DISCUSSION TITLE
Exposing addon API to popup HTML?

### DISCUSSION THREAD

**Discussion by CST1229** [ScratchAddons/ScratchAddons#3256] [Help]
How can I use the addon API in a popup's document (for example, `popups/scratch-messaging/popup.html`)? Can I use `document.addon = addon;` in the popup's JS file?
(I'm asking this because I want to add Ctrl+Enter posting to Scratch Messaging as an option)

**Comment by cobaltt7** [comment]
Ummm...why would this not be possible in the js?



________________________________
From: CST1229 ***@***.***>
Sent: Saturday, August 7, 2021 11:10:31 AM
To: ScratchAddons/ScratchAddons ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ScratchAddons/ScratchAddons] Exposing addon API to popup HTML? (#3256)


How can I use the addon API in a popup's document (for example, popups/scratch-messaging/popup.html)? Can I use document.addon = addon; in the popup's JS file?
(I'm asking this because I want to add Ctrl+Enter posting to Scratch Messaging as an option)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/ScratchAddons/ScratchAddons/discussions/3256>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AOT5DEX4C2TWLB3TRVPAY7TT3VLHPANCNFSM5BXRKIQA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.

**Reply by CST1229** [reply]
IDK I just thought of doing it in the HTML instead of the JS file (another way would be to do a `while (true)` loop with waitForElement but I don't think that will be good).

**Comment by cobaltt7** [comment]
Just use vue



________________________________

**Reply by CST1229** [reply]
I don't know how to use Vue.
(At all.)

**Reply by CST1229** [reply]
And also how do I use the addon API inside Vue?

**Comment by FunctionalMetatable** [comment]
addon.tab.waitForElement can't be used in popup, you'll have to use mutationobservers, etc

**Reply by CST1229** [reply]
I'm mostly asking for using addon.settings.get in popup, either via normal HTML or Vue somehow.

**Reply by FunctionalMetatable** [reply]
> I'm mostly asking for using addon.settings.get in popup, either via normal HTML or Vue somehow.

addon.settings.get works.
You just need to modify this function: https://github.com/ScratchAddons/ScratchAddons/blob/b887922cb2bd3894577b8e1d644c5cdc9ef616d9/popups/scratch-messaging/popup.js#L3
